### PR TITLE
[SPARK-26071][followup] Improve migration guide of disallowing map type map key

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

mention the workaround if users do want to use map type as key, and add a test to demonstrate it.

### Why are the changes needed?

it's better to provide an alternative when we ban something.

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @HyukjinKwon @zsxwing

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ license: |
 
   - The `ADD JAR` command previously returned a result set with the single value 0. It now returns an empty result set.
 
-  - In Spark version 2.4 and earlier, users can create map values with map type key via built-in function like `CreateMap`, `MapFromArrays`, etc. Since Spark 3.0, it's not allowed to create map values with map type key with these built-in functions. Users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.
+  - In Spark version 2.4 and earlier, users can create map values with map type key via built-in function like `CreateMap`, `MapFromArrays`, etc. Since Spark 3.0, it's not allowed to create map values with map type key with these built-in functions, users can use array<struct<key, value>> as map keys instead. Users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.
```
Comment (HyukjinKwon) [inline docs/sql-migration-guide.md]: no need to change but wanted to point out: "like" -> "such as"

Comment (HyukjinKwon) [inline docs/sql-migration-guide.md]: > ... built-in functions, users can use array<struct<key, value>> as map keys instead. Users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.

I would write as below:

... built-in functions. Users can use `array<struct<key, value>>` as map keys instead as a workaround. In addition, users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.

Comment (HyukjinKwon) [reply docs/sql-migration-guide.md]: Maybe .. "though they are not very useful." -> "though it is discouraged." (?).

File: docs/sql-migration-guide.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ license: |
 
   - The `ADD JAR` command previously returned a result set with the single value 0. It now returns an empty result set.
 
-  - In Spark version 2.4 and earlier, users can create map values with map type key via built-in function like `CreateMap`, `MapFromArrays`, etc. Since Spark 3.0, it's not allowed to create map values with map type key with these built-in functions. Users can still read map values with map type key from data source or Java/Scala collections, though they are not very useful.
+  - In Spark version 2.4 and earlier, users can create map values with map type key via built-in function such as `CreateMap`, `MapFromArrays`, etc. Since Spark 3.0, it's not allowed to create map values with map type key with these built-in functions. Users can use array<struct<key, value>> as map keys instead as a workaround. In addition, users can still read map values with map type key from data source or Java/Scala collections, though it is discouraged.
```
Comment (gatorsmile) [inline docs/sql-migration-guide.md]: Could we give a very simple example, how to use array<struct<key, value>> as map keys as a workaround? Also, do we have such a test case?

**Review by gatorsmile — APPROVED** [review]
LGTM pending Jenkins

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks for review, merging to master/3.0!

**Pull request closed without merging**